analyzers/src/SonarAnalyzer.Common/Rules/UseUnixEpochBase.cs

```diff
@@ -57,23 +58,29 @@ public abstract class UseUnixEpochBase<TSyntaxKind, TLiteralExpression, TMemberA
                 Language.GeneratedCodeRecognizer,
                 c =>
                 {
-                    var literalsArguments = Language.Syntax.ArgumentExpressions(c.Node).OfType<TLiteralExpression>();
-                    if (!literalsArguments.Any(x => IsValueEqualTo(x, EpochYear) || literalsArguments.Count(x => IsValueEqualTo(x, EpochMonth)) != 2))
+                    var arguments = Language.Syntax.ArgumentExpressions(c.Node);
+                    var literalsArguments = arguments?.OfType<TLiteralExpression>();
+
+                    if (literalsArguments.Any(x => IsValueEqualTo(x, EpochYear)
+                        && literalsArguments.Count(x => IsValueEqualTo(x, EpochMonth)) == 2)
+                        && CheckAndGetTypeName(c.Node, c.SemanticModel, out var name)
+                        && IsEpochCtor(c.Node, c.SemanticModel))
                     {
-                        return;
+                        c.ReportIssue(Diagnostic.Create(Rule, c.Node.GetLocation(), name));
                     }
-
-                    var type = c.SemanticModel.GetTypeInfo(c.Node).Type;
-                    if (type.IsAny(TypesWithUnixEpochField) && IsEpochCtor(c.Node, c.SemanticModel))
+                    else if (arguments.Count() == 1
+                        && ((literalsArguments.Count() == 1  && IsValueEqualTo(literalsArguments.First(), EpochTicks))
+                            || (Language.FindConstantValue(c.SemanticModel, arguments.First()) is long ticks && ticks == EpochTicks))
```

Comment:
I don't think this extra check is necessary. The previous check is more than enough. If someone gives the Epoch ticks in hexa or binary format, then I think we can live with that 😎

Reply:
The previous check is just for the hardcoded number:
```cs
_ = new DateTime(621355968000000000); // Noncompliant
```
while this check is actually looking for constants through the tracker:
```cs
private const long EpochTicks = 621355968000000000;
...
_ = new DateTime(EpochTicks);
```